backend/devices/bitbox/device.go

```diff
@@ -127,9 +128,18 @@ type Device struct {
 	// The password policy for the wallet recovery password.
 	recoveryPasswordPolicy *PasswordPolicy
 
+	// BitBox desktop app config directory.
+	// Used to read/store channel settings.
+	// TODO: merge this with backend/config.
+	channelConfigDir string
+
+	mu sync.RWMutex
```

Comment:
A de-facto convention is for the mutex to guard everything below, until a blank line. In this case, it guards `channel`, `onEvent` and `closed`.

The `mu` is one of the most common names, like `r` for request. But it can't be `channelMutex`. It would be misleading.

Having multiple mutexes leads to to a much higher complexity and more error prone. Should avoid and have as fewer as possible. Because  this mutex is RW, most of the time it's used for read-locking, so very low contention anyway. Having multiple mutexes will not change that (contention).

Re: `defer locker.RLock()()`. I was actually going to suggest to remove that package helper from the code base. I know it's temping to have it on a single line but it's more error prone and not as much readable as having lock and unlock on separate lines, like it's supposed to be :)

Reply:
> A de-facto convention is for the mutex to guard everything below, until a blank line.

Nice, wasn't aware! :+1: 

>  I was actually going to suggest to remove that package helper from the code base. I know it's temping to have it on a single line but it's more error prone and not as much readable as having lock and unlock on separate lines,

:sob: :laughing: 

I can accept it, though I am not sure I agree. Easy to miss an unlock on a separate line. `defer lock()()` pattern, if followed everywhere, is easy and readable imho.

Reply:
The biggest issue with this is that `defer lock()` would also be happily accepted by the compiler, being a valid expression.

So, it's very easy to miss and in fact it has happened many times in the past. That's why this is generally discouraged unless in very specific cases where no other way works, but even then... best is to stay away from this.

As some I know once said, "Clear is better than clever." :) https://go-proverbs.github.io

Reply:
Of course, one might forget to `defer mu.Unlock()` too but the fact is, an typical programmer's eye usually stops at `defer lock(` and goes to the beginning of the next line. You've just trained your eye because you wrote it and used it everywhere but it's very uncommon.

Reply:
In fact, when I saw `defer lock()()` for the first time my reaction was like... "wtf, why is a mutex locked on function return and probably unlocked somewhere else???!" :)